packages/backend/src/controllers/authentication.ts

```diff
@@ -340,24 +340,26 @@ export const storeOIDCRedirect: RequestHandler = (req, res, next) => {
 export const getSuccessURLWithReturnTo = (req: Request): string => {
     const url = new URL('/api/v1/oauth/success', lightdashConfig.siteUrl);
     if (req.session.oauth?.returnTo) {
-        url.searchParams.set('returnTo', req.session.oauth.returnTo);
+        url.searchParams.set(
+            'returnTo',
+            decodeURIComponent(req.session.oauth.returnTo),
```

Comment:
Why did this move?

Reply:
Oh it makes sense, parse and validate when stored, then the function loading this data doesn't need to know to decode the string